client/src/components/Design/components/Top/Top.tsx

```diff
@@ -8,12 +10,26 @@ Top.Line = Line;
 Top.EditableLine = EditableLine;
 
 export default function Top({children}: React.PropsWithChildren) {
+  const [childrenTexts, setChildrenTexts] = useState<string[]>([]);
+
+  useEffect(() => {
+    const collectedTexts: string[] = [];
+    React.Children.forEach(children, child => {
+      if (React.isValidElement(child) && child.type === Top.Line) {
+        collectedTexts.push(child.props.text);
+      }
+    });
+    setChildrenTexts(collectedTexts);
+  }, [children]);
+
```

Comment:
```suggestion

  const collectedTexts: string[] = [];
  React.Children.forEach(children, child => {
    if (React.isValidElement(child) && child.type === Top.Line) {
      collectedTexts.push(child.props.text);
    }
  });
```

useEffect를 사용하지 않고 이렇게 바로 children의 파생 상태를 이용할 수 있을 것 같아요.

Reply:
오 ㅋㅋㅋㅋㅋ 토다리랑 동일한 부분을 말씀해주셨네요! 꼭 반영해보겠습니당